core/src/main/java/org/elasticsearch/tasks/TaskManager.java

```diff
@@ -80,7 +89,15 @@ public void setTaskResultsService(TaskResultsService taskResultsService) {
      * Returns the task manager tracked task or null if the task doesn't support the task manager
      */
     public Task register(String type, String action, TaskAwareRequest request) {
-        Task task = request.createTask(taskIdGenerator.incrementAndGet(), type, action, request.getParentTask());
+        Map<String, String> headers = new HashMap<>();
```

Comment:
even though we have `http.max_header_size` I wonder if we should also add a hard limit to the total size here. We can simply accumulate the size of the keys and values for this and fail hard if is exceeded. I really don't want to hold on to MB of strings here that are literally user input. It can be simply a derivative of `http.max_header_size`?

Reply:
Makes sense! I am going to resolve conflicts and add this check.